src/test/groovy/pl/sda/rentacar/domain/rent/RentServiceIntegrationSpec.groovy

```diff
@@ -0,0 +1,297 @@
+    DepartmentService departmentService
+
+    @Autowired
+    EmployeeService employeeService
+
+    @Autowired
+    CarService carService
+
+    @Shared
+    def clientDaniel = new Client(
+            1L,
+            "Daniel",
+            "Olbrychski",
+            "kochamskutery@szabelka.eu",
+            "777888999"
+    )
+
+    @Shared
+    def departmentZadupie = new Department(
+            1L,
+            "Zadupie",
+            [] as Set,
+            [] as Set)
+
+    @Shared
+    def employeeMax = new Employee(
+            1L,
+            "Max",
+            "Wersztapen",
+            departmentZadupie)
+
+    @Shared
+    def carForFamily = new Car(
+            1L,
+            "Renaul",
+            "Espace",
+            2007,
+            CarStatus.AVAILABLE,
+            BigDecimal.valueOf(110),
+            BodyType.MINIVAN
+    )
+    @Shared
+    def departmentRequest = new DepartmentCreateRequest(
+            "Radom",
+            [] as Set,
+            [] as Set
+    )
+
+    @Shared
+    def clientRequest = new ClientCreateRequest(
+            "Otylia",
+            "Jedrzejczak",
+            "motylek@o2.pl",
+            "555777666")
+
+    @Shared
+    def clientRequest2 = new ClientCreateRequest(
+            "Michal",
+            "Zewlakow",
+            "wybitnyreprezentant@pzpn.pl",
+            "555777888")
+
+    @Shared
+    def carRequest = new CarCreateRequest(
+            "Mazda",
+            "MX-5",
+            2019,
+            BigDecimal.valueOf(100L),
+            BodyType.ROADSTER
+    )
+
+    @Shared
+    def carRequest2 = new CarCreateRequest(
+            "Volvo",
+            "C70",
+            2012,
+            BigDecimal.valueOf(100L),
+            BodyType.CONVERTIBLE
+    )
+
+    def 'should add rent'() {
+        given:
+        cleanup()
+        def departmentId = givenDepartmentExists(departmentRequest)
+        def employeeRequest = new EmployeeCreateRequest(
+                "Rimi",
+                "Kaikkonen",
+                departmentId)
+        employeeService.addEmployee(employeeRequest)
+        clientService.addClient(clientRequest)
+        carService.addCar(carRequest)
+        def request = new RentCreateRequest(
+                clientService.getALlClients().first().getId(),
+                employeeService.findAllEmployees().first().getId(),
+                carService.getALlCars().first().getId(),
+                "xD")
+        service.addRent(request)
+
+        when:
+        def result = repository.findAll().first()
+
+        then:
+        with(result) {
+            id != null
+            client.id == request.clientId
+            employee.id == request.employeeId
+            car.id == request.carId
+            startDate == LocalDate.now()
+            returnDate == null
+            charge == null
+            comment == request.comment
+            rentStatus == RentStatus.ACTIVE
+        }
+    }
+
+    def 'should map Rent to RentView'() {
```

Comment:
This is a unit test and it shouldn't be here. It should be placed in `RentMapperSpec`. Also all methods in `RentMapper` should be tested.